book_haven_api/tests/requests/libraries/books/test_add_book_to_library.py

```diff
@@ -0,0 +1,60 @@
+import pytest
+from django.urls import reverse
+from rest_framework.test import APIClient
+from book_haven_api.models import Library, Book
+
+@pytest.fixture
+def Libraries():
+    library1 =  Library.objects.create(
+        id=1,
+        name='Library1',
+        street='1234 Library1 St.',
+        city='Library1 City',
+        state='CA',
+        zip='12345',
+        lat=123.456,
+        lon=123.456,
+    )
+
+    return library1
+
+
+@pytest.mark.django_db
+def test_add_book_to_library(Libraries):
+    print('add_book_to_library_happy_path')
+    library1 = Libraries
+    url = reverse('get_library_books', kwargs={'library_id': library1.id})
+    client = APIClient()
+    response = client.post(url, {'isbn': '9780446675505'}, format='json')
+    assert response.status_code == 201
+    assert type(response.json()) is dict
+    assert type(response.json()['data']) is dict
+    attrs = response.json()['data']['attributes']
+    assert type(attrs) is dict
+    assert attrs['isbn'] == '9780446675505'
+    assert attrs['book_image'] == 'http://books.google.com/books/content?id=OemtHIwYk0wC&printsec=frontcover&img=1&zoom=1&source=gbs_api'
+    assert attrs['description'] == 'In a futuristic society filled with chaos, young Lauren Olamina begins a journey that will test her will and ultimately start a new faith. Includes questions for discussion.'
+    assert attrs['title'] == 'Parable of the Sower'
+    assert attrs['author'] == 'Octavia E. Butler'
+    assert attrs['genre'] == 'FICTION'
+    assert attrs['library_id'] == 1
```

Comment:
In general, once you move to having the ID be auto generated, we don't really care about testing ID's - if you want to test order of something, for example, you'd usually use a `created_at` or something similar. It's ok to use ID in your tests for sure, just typically in rare cases.

Reply:
Great point.